src/main/java/uk/gov/digital/ho/proving/income/api/FinancialStatusService.java

```diff
@@ -46,34 +46,50 @@ public FinancialStatusService(HmrcClient hmrcClient, AuditClient auditClient, Ni
         this.ninoUtils = ninoUtils;
     }
 
+    @Deprecated
     @PostMapping(value = "/incomeproving/v2/individual/financialstatus", produces = APPLICATION_JSON_VALUE)
+    public FinancialStatusCheckResponseV2 getFinancialStatusV2(@Valid @RequestBody FinancialStatusRequestV2 requestV2) {
+        Applicant applicant = new Applicant(requestV2.getForename(), requestV2.getSurname(), requestV2.getDateOfBirth(), requestV2.getNino());
+        FinancialStatusRequest request = new FinancialStatusRequest(Arrays.asList(applicant), requestV2.getDateOfBirth(), requestV2.getDependants());
+
+        FinancialStatusCheckResponse response = getFinancialStatus(request);
+
+        CategoryCheck categoryCheck = response.categoryChecks().get(0);
+        CategoryCheckV2 categoryCheckV2 = new CategoryCheckV2(categoryCheck.category(), categoryCheck.passed(), categoryCheck.failureReason(), categoryCheck.applicationRaisedDate(), categoryCheck.assessmentStartDate(), categoryCheck.threshold(), categoryCheck.individuals().get(0).employers());
+        FinancialStatusCheckResponseV2 responseV2 = new FinancialStatusCheckResponseV2(response.status(), response.individuals().get(0), categoryCheckV2);
+        return responseV2;
+    }
+
+    @PostMapping(value = "/incomeproving/v3/individual/financialstatus", produces = APPLICATION_JSON_VALUE)
     public FinancialStatusCheckResponse getFinancialStatus(@Valid @RequestBody FinancialStatusRequest request) {
 
-        final String redactedNino = ninoUtils.redact(request.getNino());
-        log.info("Financial status check request received for {} - applicationRaisedDate = {}, dependents = {}", redactedNino, request.getApplicationRaisedDate(), request.getDependants());
+        Applicant mainApplicant = request.applicants().get(0);
```

Comment:
The request is checked with @Valid - should the class have a not null validation on applicants to make this statement null safe? Is it possible to be called with no applicants?

Reply:
Agreed - I will take this up on the next piece of dev.